rt/management/src/main/java/org/apache/cxf/management/jmx/MBServerConnectorFactory.java

```diff
@@ -169,18 +174,71 @@ public void run() {
         } else {
              // Start the connector server in the same thread.
             connectorServer.start();
+            remoteServerStub = rmiServer.toStub();
         }
 
         if (LOG.isLoggable(Level.INFO)) {
             LOG.info("JMX connector server started: " + connectorServer);
         }
     }
 
+    static int getRegistryPort(final String url) {
+        int serverStart = url.indexOf("/jndi/rmi://");
+        final String serverPart = url.substring(serverStart + 12);
+        int portStart = serverPart.indexOf(':') + 1;
+
+        int portEnd;
+        int port = 0;
+        if (portStart > 0) {
+            portEnd = indexNotOfNumber(serverPart, portStart);
+            if (portEnd > portStart) {
+                final String portString = serverPart.substring(portStart, portEnd);
+                port = Integer.parseInt(portString);
+            }
+        }
+        return port;
+    }
+
     public void destroy() throws IOException {
         connectorServer.stop();
         if (LOG.isLoggable(Level.INFO)) {
             LOG.info("JMX connector server stopped: " + connectorServer);
         }
     }
 
+    /*
+     * Better to use the internal API than re-invent the wheel.
+     */
+    @SuppressWarnings("restriction")
+    private class JmxRegistry extends sun.rmi.registry.RegistryImpl {
+        static final String LOOKUP_NAME = "jmxrmi";
```

Comment:
I'm just wondering what happens if we define a URL that doesn't end with jmxrmi....shouldn't we take the lookup name from the serviceUrl?

Reply:
Good point. Let's see if we can do that in a smarter way.

Reply:
Hopefully done in the latest commit.

Reply:
Looks good thanks